### PR TITLE
Persistent entropy

### PR DESCRIPTION
**Pull request by EduPH**
First implementation of persistent entropy.

* "persistent_entropy.py" was added.

Pending: Tutorial notebook.

### CODE REVIEW

**Comment by sauln** [comment]
Looks great. 

Once a few unit tests and some documentation are in place, I'll add a code review and test out the docs.

File: .gitignore [hunk 1]
```diff
@@ -8,4 +8,5 @@ __pycache__
 .ipynb_checkpoints
 
 build
-dist
\ No newline at end of file
+dist
+docs/reference/.#index.rst
```
Comment (sauln) [inline .gitignore]: It looks like the `.#` file is from your text editor. Could you please remove this from the PR.

File: docs/reference/index.rst [hunk 2]
```diff
@@ -36,3 +36,15 @@ Diagram Visualization
     persim.plot_diagrams
     persim.bottleneck_matching
     persim.wasserstein_matching
+
+
+
+    
+Persistence barcode measure
+======================
```
Comment (sauln) [inline docs/reference/index.rst]: Have you tried rendering the docs? I think you'll need to add more "=" until there are enough to completely cover the text.

File: persim/persistent_entropy.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+"""
+
+    Implementation of persistent entropy
```
Comment (sauln) [inline persim/persistent_entropy.py]: Please add a reference paper that first proposed the entropy here.

File: persim/persistent_entropy.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    Author: Eduardo Paluzo Hidalgo (cimagroup, University of Seville)
+    contact: epaluzo@us.es
+
+"""
+
+import numpy as np
+
+
+__all__ = ["persistent_entropy"]
```
Comment (sauln) [inline persim/persistent_entropy.py]: this string should match the function name. I'd prefer the function name to be `persistent_entropy` like the string says.

File: persim/persistent_entropy.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+def pentropy(dgms, inf = 0, valInf = -1, norm = 0):
```
Comment (sauln) [inline persim/persistent_entropy.py]: the `inf` variable is a boolean. Please change it to `True` or `False` and make the variable name reflect this. Maybe `keep_inf=False` by default?

Comment (sauln) [inline persim/persistent_entropy.py]: Please use snake case for `valInf` -> `val_inf`.  Have the default be `None` instead.

Comment (sauln) [inline persim/persistent_entropy.py]: Use boolean explicitly for `norm`, and please change to a verb. I think the full word `normalize` would make more sense.

File: persim/persistent_entropy.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    Perform the persistent entropy values of a family of persistence barcodes (or persistence diagrams).
+    Assumes that the input diagrams are from a determined dimension. If the infinity bars have any meaning
+    in your experiment and you want to keep them, remember to give the value you desire to valInf.
+
+    Parameters
+    -----------
+    dgms: array of arrays of birth/death pairs of a persistence barcode of a determined dimension.
```
Comment (sauln) [inline persim/persistent_entropy.py]: For docstrings, please use the following convention:
```
variable_name: type(variable_name)
    description of the purpose of the variable
```

File: persim/persistent_entropy.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    # Step 1: Remove infinity bars if inf = 1. If inf = 0, infinity value is substituted by valInf.
+
+    if inf == 0:
+        dgms = [(dgm[dgm[:,1] !=np.inf]) for dgm in dgms]
+    if inf == 1:
+        if valInf !=-1:
+            dgms =  [np.where(dgm==np.inf,valInf,dgm) for dgm in dgms]
+        else:
+            print("Remember: You need to provide a value to infinity bars if you want to keep them.")
```
Comment (sauln) [inline persim/persistent_entropy.py]: If it gets to this print statement, the rest of the code should not run, right? How about raising an error instead?

File: persim/persistent_entropy.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            
+            
+        
+    # Step 2: Persistent entropy computation.
+    ps = []
+    for dgm in dgms:
```
Comment (sauln) [inline persim/persistent_entropy.py]: What happens if only one diagram is input? Please add some handling for this scenario and a test to make sure everything works alright.

In other places in scikit-tda, we've taken a single diagram or a list of diagrams. Try looking at the `plot_diagrams` code for an example of how we've handled it in the past. Not sure if it's the best method though, maybe you have better ideas?

File: persim/persistent_entropy.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        L = np.sum(l)
+        p = l/L
+        E = -np.sum(p*np.log(p))
+        if norm ==1:
+            E = E/np.log(len(l))
+        ps.append(E)
+
+    return np.array(ps)
+
+
```
Comment (sauln) [inline persim/persistent_entropy.py]: Please apply the [Black formatter](https://black.readthedocs.io/en/stable/) to this file.

**Comment by EduPH** [comment]
"SyntaxError: Non-ASCII character '\xe2' in file /home/travis/build/scikit-tda/persim/persim/persistent_entropy.py" in python 2.7... Not really sure about how to manage that issue.

**Pull request merged**